File: src/main/kotlin/org/quintilis/factions/commands/ClanCommands.java
package org.quintilis.factions.commands;

public enum ClanCommands implements CommandInterface {

    CREATE("create", "/clan create <name> <tag>"),
    DELETE("delete", "/clan delete"),
    LIST("list", "/clan list"),
    SET("set", "/clan set <field> <value>"),
    MEMBER("member", "/clan member <invite|kick|list>"),
    QUIT("quit", "/clan quit");

    private final String command;
    private final String usage;

    ClanCommands(String command, String usage) {
        this.command = command;
        this.usage = usage;
    }

    public String getCommand() {
        return this.command;
    }
    public String getUsage() {
        return this.usage;
    }

}

File: src/main/kotlin/org/quintilis/factions/commands/clan/ClanCommands.kt
package org.quintilis.factions.commands.clan

import org.quintilis.factions.commands.Commands
import org.quintilis.factions.commands.HelpEntry

enum class ClanCommands(
    override val command: String,
    override val usage: String,
    override val helpEntry: HelpEntry,
    override val subCommands: Array<out Commands>? = null
): Commands {
    CREATE(
        "create",
        "/clan create [name] [tag]",
        HelpEntry(
            "clan.create.description",
            "factions.usage"
        )
    ),
    DELETE(
        "delete",
        "/clan delete",
        HelpEntry(
            "clan.delete.description",
            "factions.usage"
        )
    ),
    LIST(
        "list",
        "/clan list <page>",
        HelpEntry(
            "clan.list.description",
            "factions.usage"
        )
    ),
    ALLY(
        "ally",
        "/clan ally <subcommand>",
        HelpEntry(
            "clan.ally.description",
            "factions.usage"
        ),
        AllySubCommands.entries.toTypedArray()
    ),
    QUIT(
        "quit",
        "/clan quit",
        HelpEntry(
            "clan.quit.description",
            "factions.usage"
        )
    ),
    MEMBER(
        "member",
        "/clan member <subcommand>",
        HelpEntry("clan.member.description", "factions.usage"),
        MemberSubCommands.entries.toTypedArray()
    ),
    INVITE(
        "invite",
        "/clan invite <subcommand>",
        HelpEntry("clan.invite.description", "factions.usage"),
        InviteSubCommands.entries.toTypedArray()
    )
}